pluginLauncher/lib/source/DobbyRdkPluginManager.cpp

```diff
@@ -296,6 +297,74 @@ void DobbyRdkPluginManager::loadPlugins()
     AI_LOG_FN_EXIT();
 }
 
+// -----------------------------------------------------------------------------
+/**
+ *  @brief Prepares the dependency solver and required plugins data structures.
+ *
+ *  This method scans the container config and based on its contents:
+ *  1. Adds all the plugins, along with their dependencies, to the plugin dependency solver,
+ *  2. Creates a list of the required plugins,
+ *  3. Checks if the required plugins are loaded.
+ *
+ *  @return False if a required plugin is not loaded or if one of the dependencies is not a known plugin.
+ *  True otherwise.
+ */
+bool DobbyRdkPluginManager::preprocessPlugins()
+{
+    AI_LOG_FN_ENTRY();
+
+    const auto pluginsInConfig = mContainerConfig->rdk_plugins->names_of_plugins;
+    const int rdkPluginCount = mContainerConfig->rdk_plugins->plugins_count;
+
+    // Add plugins to the solver, remember which ones are required.
+    for (size_t i = 0; i < rdkPluginCount; i++)
+    {
+        const std::string pluginName = pluginsInConfig[i];
+        const bool required = mContainerConfig->rdk_plugins->required_plugins[i];
+        if (required)
+        {
+            mRequiredPlugins.insert(pluginName);
+        }
+
+        mDependencySolver.addPlugin(pluginName);
+    }
+
+    // Check if required plugins are loaded, store plugin dependencies.
+    for (size_t i = 0; i < rdkPluginCount; i++)
+    {
+        const std::string pluginName = pluginsInConfig[i];
+        if (!isLoaded(pluginName))
+        {
+            if (isRequired(pluginName))
+            {
+                AI_LOG_ERROR_EXIT("Required plugin %s isn't loaded, but present in the container config - aborting", pluginName.c_str());
+                return false;
```

Comment:
This method returns false but the return value is never used/checked as this is called from the constructor.

Potential idea - add an `isValid` flag that is set to `false` in these scenarios (and possibly during the `loadPlugins()` method as well, then other methods can check this flag to check if plugin loading & pre-processing was successful and fail fast if not.

Reply:
Good idea, I've added the mValid flag.